### PR TITLE
Add mega65 target

### LINKED ISSUE #44: MEGA65 target
I have set up a [small project](https://github.com/mlund/mega65-adventure) to use llvm-mos-sdk with rust with the aim to run on the [Mega65](https://www.mega65.org) computer. This has a C64 compatibility mode (`go 64` a la C128) where the test program runs fine. I wonder what is required to make a true Mega65 target? I'm aware that the mos-llvm project currently doesn't generate CSG4510 instructions (based on 65CE02), but that's perhaps not initially a problem as it's 6502 compatible. I have started a [fork](https://github.com/mlund/llvm-mos-sdk) with what I think are the structural parts, but so far it's merely a clone of the C64 target. Any input would be highly appreciated.

### PR DESCRIPTION
**Pull request by mlund**
This adds a MEGA65 target and should further be compatible with the unreleased Commodore 65 computer. This is a basic but functional implementation; memory regions could likely be optimised to better use the expanded memory of the mega. Closes #44.

- [x] Add mega65 target
- [x] Update README 
- [x] Resolve issue with `printf` (`examples/hello-new`)

~~Memory is allocated in $2001-$BFFF and `__stack` is set to $BFFF. However, this seem to give problems with `printf` that prints the same number regardless of input. Lowering the stack location solves this issue.~~
Currently the SYS command does not map bank 0's $8000-$C000 region which caused the above issue. Ping @mysterymath

### CODE REVIEW

File: mos-platform/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ add_subdirectory(common)
 add_subdirectory(atari8)
 add_subdirectory(commodore)
 add_subdirectory(c64)
+add_subdirectory(mega65)
```
Comment (mlund) [inline mos-platform/CMakeLists.txt:41]: Sort alphabetically?

File: mos-platform/mega65/basic-header.S [hunk 2]
```diff
@@ -0,0 +1,10 @@
+; MEGA65 BASIC header that jumps to _start.
+
+.section .basic_header,"aR",@progbits
+#	.short next_line	; address of next BASIC line
+        .byte 0x0a, 0x20, 0x0a, 0x00, 0xfe, 0x02, 0x20, 0x30, 0x00 ; 10 BANK 0
+        .byte 0x15, 0x20, 0x14, 0x00, 0x9e, 0x20 ; 20 SYS
+	.mos_addr_asciz _start, 4	; the location of the _start symbol, in 4 decimal
```
Comment (mlund) [inline mos-platform/mega65/basic-header.S:7]: is there a more elegant way of doing this?

Comment (mysterymath) [reply mos-platform/mega65/basic-header.S:7]: Not that I know of; this seems pretty idiomatic for BASIC headers.

**Review by mysterymath — APPROVED** [review]
Looks good to me!

**Pull request merged**